test/sql/types/nested/list/test_list_reverse.test

```diff
@@ -0,0 +1,67 @@
+# name: test/sql/types/nested/list/test_list_reverse.test
+# description: Test reversing the order of the elements of a list
+# group: [list]
+
+statement ok
+PRAGMA enable_verification
+
+query I
+SELECT LIST_REVERSE(ARRAY[NULL])
+----
+[NULL]
+
+query I
+SELECT LIST_REVERSE(ARRAY[])
+----
+[]
+
+query I
+SELECT LIST_REVERSE(ARRAY[1, 42, 2])
+----
+[2, 42, 1]
+
+query I
+SELECT LIST_REVERSE(ARRAY[1, 42, NULL, 2])
+----
+[2, NULL, 42, 1]
+
+query I
+SELECT LIST_REVERSE (LIST_REVERSE(ARRAY[1, 3, 3, 42, 117, 69, NULL]))
```

Comment:
Could we add some tests with data in a table? e.g.:

```sql
CREATE TABLE lists(l INT[]);
INSERT INTO lists VALUES ([1, 2, 3]), (NULL), ([]), ([NULL, NULL, 5, 6, NULL]);
SELECT list_reverse(l) FROM lists;
```

And tests with `test_vector_types` and `test_all_types`, e.g.::

```sql
select list_reverse(columns(['int_array', 'varchar_array', 'nested_int_array', 'array_of_structs', 'timestamp_array', 'double_array', 'date_array', 'timestamptz_array'])) from test_all_types();
select list_reverse(test_vector) from test_vector_types(null::int[], false);
select list_reverse(test_vector) from test_vector_types(null::int[], true);
```